MewsSystems/mews-flutter
# fix(optimus): [RND-124891] Date picker not taking input on Android

RND-124891

#### Summary

- Fixed `DateInput` component not taking the input on Android. (It's a Flutter bug with the `TextInputValue.number`). Found a workaround for that.
- Fixed value of the `DateInput` being reset to the `initValue` on the rebuild. 

#### Testing steps

Since the component by itself was not showing this bug in the Storybook, it needs to be tested inside the Kiosk reservation details screen using Android.

1.  Open the reservation screen in `Kiosk` on the Android.
2. `ClearAll` button should remove the initValue and not set back to it when clicked on.
3. Input for this component should work as intended.

#### Follow-up issues

None

#### Check during review

- Verify against YouTrack issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).
